079-removal-of-mirror-maker-1.md

```diff
@@ -0,0 +1,131 @@
+And therefore the point when the Mirror Maker 1 support will be removed is not obvious.
+Will it be removed when right away Kafka 4.0 is adopted?
+Or only once support for Kafka 3.x is completely dropped?
+
+This proposal aims to follow up on the [proposal #77](https://github.com/strimzi/proposals/blob/main/077-support-for-kafka-4.0.md) which discussed the adoption of Kafka 4.0 and support for KRaft and ZooKeeper based clusters.
+It formalizes the removal timeline for Mirror Maker 1 and clarifies what will be done by Strimzi and what will be done by the users.
+It does not aim to provide any step-by-step migration guide.
+
+## Expectations
+
+This proposal uses the same expectations as the [proposal #77](https://github.com/strimzi/proposals/blob/main/077-support-for-kafka-4.0.md) with regards to Kafka releases:
+* Apache Kafka 3.8.0 release (currently in the release process, expected release in July or August?)
+* Apache Kafka 3.9.0 release with the remaining missing KRaft features (shorter release cycle - expected August or September?)
+* Apache Kafka 4.0.0 release 3-4 months after the Kafka 3.9 release without Mirror Maker 1 (late 2024, early 2025?)
+
+## Proposal
+
+Strimzi will remove support for Mirror Maker 1 together with ZooKeeper support.
+This would mean that the last version supporting Mirror Maker 1 will be 0.4y.
+This is also the last version with ZooKeeper support and the version selected for the extended support in [proposal #77](https://github.com/strimzi/proposals/blob/main/077-support-for-kafka-4.0.md).
+Support for Mirror Maker 1 will be dropped in Strimzi 0.4z.
+
+The following table shows how the Mirror Maker 1 (MM1) support fits with Kafka 4.0, ZooKeeper, and Kraft support.
+
+| Version | Supported Kafka versions | Supported metadata modes | Extended support | MM1 support |
+| :-----: |:-------------------------|:-------------------------|:-----------------|:------------|
+| ...     | 3.7.x, 3.8.x             | ZooKeeper, KRaft         | no               | yes         |
+| 0.4x    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         | no               | yes         |
+| 0.4y    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         | **yes**          | yes         |
+| 0.4z    | 3.9.x, 4.0.x             | KRaft                    | no               | no          |
+| ...     | 4.0.x, 4.1.x             | KRaft                    | no               | no          |
+
+Users will be responsible for migrating their Mirror Maker 1 clusters to Mirror Maker 2 (or any other alternative they might choose) before upgrading to Strimzi 0.4z.
+While it is possible to configure Mirror Maker 2 to closely resemble the functionality of Mirror Maker 1, this proposal does not cover the detailed steps of how users should migrate from Mirror Maker 1 to Mirror Maker 2.
+
+If the Mirror Maker 1 cluster is not migrated before upgrade to Strimzi 0.4z or later, Strimzi will stop operating it.
+But it will keep it running and not delete it.
```

Comment:
I think this sentence could be clearer. There are two 'it's which mean different things. I think we're saying "Strimzi will stop managing (but not delete) the MM1 CR or `Deployment`, so the MM1 Pod will continue to run."

Reply:
You are right, this is a bit confusing. I reworded it to make it (hopefully) more clear.